### DISCUSSION TITLE
PyCaret: overriding inferred data types is not working?

### DISCUSSION THREAD

**Discussion by tonpimenta** [pycaret/pycaret#3837] [Q&A] [answered]
I'm trying initialize the` setup() `function specifying` categorical_features` and `numeric_features` to **prevent** that the algorithm automatically infer some of my columns as Ordinal Features. However, still the output of the `setup()` function gives me the same amount of columns as Ordinal Features anyway. Does anyone know how to successfully override these data types? I'm concerned this is something specific to the fact I'm using Google Colab.
Here's the code:

```
from pycaret.classification import *
data2pycaret = df.join(df_cat)
#temp = df_cat.drop[columns = 'TARGET']
cat_list = df_cat.drop(columns = 'TARGET').columns.tolist()
num_list = df.columns.tolist()
print(cat_list)
print(num_list)
s = setup(data2pycaret, target = 'TARGET', session_id = 125,
          preprocess = True, 
          feature_selection = True, 
          fix_imbalance = True,
          normalize = True, 
          categorical_features = cat_list, 
          numeric_features = num_list,
          ordinal_features = None,
          max_encoding_ohe=25)
```

Here's the output I'm getting:
![Capturar](https://github.com/pycaret/pycaret/assets/97309305/e3f372b2-e553-4769-b437-32cd9a062db6)

**Comment by celestinoxp** [comment]
Check Data Types:
Ensure that the columns in cat_list are indeed of categorical type, and the columns in num_list are of numeric type. Sometimes, data can be misread.

Execution Order:
In the provided code, it appears that df and df_cat are being joined, but we don't have visibility into the contents of these DataFrames. Ensure that the joining is happening as expected.

Version of PyCaret:
Check if the user is using a compatible version of PyCaret. It might be helpful to update to the latest version if not.

Restart Environment:
In Google Colab, it might be beneficial to restart the environment and run the code again to ensure there are no lingering issues.

Consider read the docs: https://pycaret.readthedocs.io/en/stable/api/classification.html

**Reply by tonpimenta** [reply]
Thanks for the willingness to help.
And to update, I tried do move from Colab to Jupyter Notebook and installed PyCaret from scratch on my work's computer. I had the exact same result.
I'm not a programming expert, but I'll try to provide the best answers. So:
- Execution Order:
df and df_cat are being joined accordingly.
- Version of Pycaret:
Latest indeed
- Restarted Env:
Several times
- Data types:
From what I understand, yes, the columns in `cat_list` are categorical and the ones at `num_list`, numeric. Here's the full code:
```
df = pd.read_excel("MGMTmet_UPENN_PYRAD+QUALI.xlsx")
df = df.drop(columns=['ID'])
counter = 0
integer_columns = df.select_dtypes(include='int64')
for col in integer_columns:
    if  df[col].nunique() <= 5:
        counter = counter + 1
        df[col] = df[col].astype('category')

df_cat = df.select_dtypes(include='category')
df = df.select_dtypes(include='number')

from pycaret.classification import *
data2pycaret = df.join(df_cat)

cat_list = df_cat.drop(columns = 'TARGET').columns.tolist()
num_list = df.columns.tolist()

s = setup(data2pycaret, target = 'TARGET', session_id = 125,
          preprocess = True,
          feature_selection = True,
          fix_imbalance = True,
          normalize = True,
          categorical_features = cat_list,
          numeric_features = num_list,
          ordinal_features= None,
          max_encoding_ohe=25)
```
The result is the same of my original screen shot.
When I try to look which features it is considering as ordinals and categoricals:
```
print(f'Ordinal features: {s._fxs["Ordinal"]}')
print(f'Categorical features: {s._fxs["Categorical"]}')
```
I have this result:
```
Ordinal features: {'TF_RF13': [1, 2], 'CF6': [1, 2], 'CF9': [0, 2], 'CF10': [1, 2], 'CF11': [1, 2], 'CF12': [1, 2], 'CF13': [1, 2]}
Categorical features: ['CF13b', 'T_RF13', 'TF_RF6', 'TF_RF13', 'F_RF6', 'F_RF13', 'F_RF40', 'CF1', 'CF2', 'CF3', 'CF4', 'CF5', 'CF6', 'CF7', 'CF8', 'CF9', 'CF10', 'CF11', 'CF12', 'CF13']

```
Note that all ordinal features are contained within the categorical group as well. Is this supposed to happen?
I'm setting  `ordinal_features= None`, so why it is still inferring some features as ordinals?
Thanks again! 
Any help is much appreciated

**Comment by tonpimenta** [comment, marked as answer]
Question resolved by [#3848 ](https://github.com/pycaret/pycaret/pull/3848)
